source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/workflow/app.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import { Route } from 'react-router-dom';
+import { DWKitForm } from "@assets/vendor/dwkit/optimajet-form.js";
+import {
+  ApplicationRouter, NotificationComponent, FormContent,
+  FlowContent, Thunks, Store, Actions, SignalRConnector, StateBindedForm, API
+} from '@assets/vendor/dwkit/optimajet-app.js';
+
+import { compose } from 'recompose';
+import { requireAuth } from '@lib/auth';
+import { withCurrentUser } from '~/data/containers/with-current-user';
+import { withLayout } from '@ui/components/layout';
+import { attributesFor } from '~/data';
+import * as toast from '@lib/toast';
+
+import '~/global-config';
+
+window.alertify = {
+  error(...args) {
+    toast.error(...args);
+  },
+  confirm(...args) {
+    toast.info(...args);
+  },
+  log(...args) {
+    toast.info(...args);
+  },
+  success(...args) {
+    toast.success(...args);
+  },
+  set() { },
+  custom() { },
+  prompt() { },
+};
+
+window.Pace = {
+  start() { },
+  stop() { },
+};
+
+
+
+export class App extends React.Component {
+  constructor(props) {
+    super(props);
+    console.log('jQuery', $);
+    this.state = {
+      pagekey: 0
+    };
+
+    const me = this;
+
+    window.DWKitApp = this;
+    window.DWKitApp.API = API;
+    this.onFetchStarted();
+  }
+
+  render() {
+    const { currentUser } = this.props;
+
+    const sectorprops = {
+      eventFunc: this.actionsFetch.bind(this),
+      getAdditionalDataForControl: this.additionalFetch.bind(this, undefined)
+    };
+
+    const state = Store.getState();
+    console.log(Store, state);
+
+    let user = state.app.user;
```

Comment:
it's better do `let user = state.app.user || {}`

Reply:
not my code.